junit-platform-engine/src/main/java/org/junit/platform/engine/UniqueId.java

```diff
@@ -82,10 +83,14 @@ public static UniqueId root(String segmentType, String value) {
 
 	private final UniqueIdFormat uniqueIdFormat;
 	private final List<Segment> segments;
+	private transient int hashCode;
+	private transient String toString;
```

Comment:
Transient in order to keep former serialized objects de-serializable?

Reply:
Yes and setting them in `readObject`.

Reply:
Wonder if we have to go this route. As the expected life-cycle of a serialized `UniqueId` object is short. Like within an IDE-based testing session. @junit-team/junit-lambda what do think?